youtube_dl/extractor/cda.py

```diff
@@ -0,0 +1,80 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+import re
+
+from .common import InfoExtractor
+from ..utils import (
+    decode_packed_codes,
+    ExtractorError,
+    parse_duration,
+)
+
+
+class CDAIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:(?:www|ebd)\.)?cda\.pl/(?:video|[0-9]+x[0-9]+)/(?P<id>[0-9a-z]+)'
+    _TESTS = [
+        {
+            'url': 'http://www.cda.pl/video/5749950c',
+            'md5': '6f844bf51b15f31fae165365707ae970',
+            'info_dict': {
+                'id': '5749950c',
+                'ext': 'mp4',
+                'height': 720,
+                'title': 'Oto dlaczego przed zakrętem należy zwolnić.',
+                'duration': 39
+            }
+        },
+        {
+            'url': 'http://www.cda.pl/video/57413289',
+            'md5': 'a88828770a8310fc00be6c95faf7f4d5',
+            'info_dict': {
+                'id': '57413289',
+                'ext': 'mp4',
+                'title': 'Lądowanie na lotnisku na Maderze',
+                'duration': 137
+            }
+        }
+    ]
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        webpage = self._download_webpage('http://ebd.cda.pl/0x0/' + video_id, video_id)
+
+        if 'Ten film jest dostępny dla użytkowników premium' in webpage:
+            raise ExtractorError('This video is only available for premium users.', expected=True)
+
+        title = self._html_search_regex(r'<title>(.+?)</title>', webpage, 'title', fatal=False)
+
+        def _get_format(page):
+            unpacked = decode_packed_codes(page)
+            duration = self._search_regex(r"duration:\\'(.+?)\\'", unpacked, 'duration', default=None)
+            format_id = None
+            height = None
+
+            m = re.search(r'<a data-quality="(?P<format_id>[^"]+)" href="[^"]+" class="quality-btn quality-btn-active">(?P<height>[0-9]+)p<\/a>', page)
+            if m:
+                format_id = m.group('format_id')
+                height = int(m.group('height'))
+
+            return {
+                'duration': parse_duration(duration),
+                'format_id': format_id,
+                'height': height,
+                'url': self._search_regex(r"url:\\'(.+?)\\'", unpacked, 'url')
+            }
+
+        formats = [_get_format(webpage)]
+
+        pattern = re.compile(r'<a data-quality="[^"]+" href="([^"]+)" class="quality-btn">([0-9]+p)<\/a>')
+        for version in re.findall(pattern, webpage):
+            webpage = self._download_webpage(version[0], video_id, 'Fetching %s version information' % version[1])
```

Comment:
Not sure what general strategy is. But if this one fails for any reason it means that whole extraction is not good. We know for sure that the format is available and the page is available. So only thing that can really fail if it will be intermediate error. I personally don''t see a point in returning incomplete results instead of error in case of such.